python/lsst/ctrl/mpexec/singleQuantumExecutor.py

```diff
@@ -387,7 +343,14 @@ def findOutputs(
                 checkRefs: list[DatasetRef] = []
                 registryRefToQuantumRef: dict[DatasetRef, DatasetRef] = {}
                 for datasetRef in datasetRefs:
-                    ref = self._resolve_ref(datasetRef, collections)
+                    if self.butler is not None:
+                        # If running with full butler check registry.
+                        ref = self.butler.registry.findDataset(
+                            datasetRef.datasetType, datasetRef.dataId, collections=collections
+                        )
```

Comment:
I don't understand why we'd want to re-resolve here.  What could lead to there being a dataset with a different UUID/run in the full-butler case that is impossible in the QBB case?

Reply:
It's not re-resolving, but merely checking for existing refs. This is the logic that implements skip-existing-in checks for `pipetask run`. I'm actually not very happy to have this logic here, but we have that option enabled for `pipetask run`. Maybe we should disable that option together with `output-run` and only have them enabled for quantum graph generation?

Reply:
I had thought it was already only an option for `pipetask run` in the case where it was also making a QG, which I think it consistent with disabling that option when using an existing QG and removing this check

Reply:
I'll add a check to CLI to issue a warning when skip-existing is set with existing QG. This code is also used to find datasets that will be clobbered if clobbering is requested. I think I can modify it to use more efficient `Datastore.mexist()` instead of querying registry with DataIds.

Reply:
Discussion on [DM-32940](https://jira.lsstcorp.org/browse/DM-32940) may be relevant for that `mexists` attempt.  I suspect `mexists` (like `datasetExists` may be defined in a not-quite-substitutable way between `LimitedButler` and `Butler`, and fixing that may be the final piece of the puzzle in being able to unify SingleQuantumExecutor and PreExecinit implementations for the different butler types.

Reply:
Don't both butlers use a real datastore though? So datastore.knows_these and datastore.mexists should work in both cases.

Reply:
Yes, I think that's right - using datastore directly would circumvent the need for a butler API that always does only that.